xchange-okcoin/src/main/java/info/bitrich/xchangestream/okcoin/OkCoinStreamingService.java

```diff
@@ -4,16 +4,44 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import info.bitrich.xchangestream.okcoin.dto.WebSocketMessage;
 import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.reactivex.Completable;
+import io.reactivex.CompletableSource;
 import org.knowm.xchange.exceptions.ExchangeException;
 
 import java.io.IOException;
+import java.util.Timer;
+import java.util.TimerTask;
 
 public class OkCoinStreamingService extends JsonNettyStreamingService {
 
+    private Timer pingPongTimer = null;
+
     public OkCoinStreamingService(String apiUrl) {
         super(apiUrl);
     }
 
+    @Override
+    public Completable connect() {
+        Completable conn = super.connect();
+        return conn.andThen((CompletableSource)(completable) -> {
+            try {
+                if (pingPongTimer != null) {
+                    pingPongTimer.cancel();
+                }
+                pingPongTimer = new Timer("OkexPingPong", false);
```

Comment:
Usage of reactive timer would be nicer here

Reply:
Has been changed in the newest commit.

Also, the ping-pong subscription will be disposed now when the ws is disconnected.